### PR TITLE
Pyupgrade and blacken-docs

### PR DESCRIPTION
**Pull request by benjeffery**
These are the last two additions to bring tsinfer inline with tskit.
Stacked on #385

### CODE REVIEW

**Review by jeromekelleher — APPROVED** [review]

**Comment by benjeffery** [comment]
Don't merge - using this PR to inspect python build flags on OSX.

File: _tsinfermodule.c [hunk 1]
```diff
@@ -547,7 +547,7 @@ TreeSequenceBuilder_add_path(TreeSequenceBuilder *self, PyObject *args, PyObject
         goto out;
     }
     shape = PyArray_DIMS(right_array);
-    if (shape[0] != (npy_intp) num_edges) {
```
Comment (jeromekelleher) [inline _tsinfermodule.c]: Looks like something happened during rebase here - I think I added these casts a few days ago to get rid of compiler warnings.

**Review by jeromekelleher — COMMENTED** [review]
Looks good @benjeffery, I think I spotted a driveby change on the C code though.

File: _tsinfermodule.c (hunk 1 above)
Comment (benjeffery) [reply _tsinfermodule.c]: Yeah I spotted it too - think it was while I was messing around trying to find the segfault. Should be all cleared up now as I hard reset and re-ran pre-commit.

**Comment by jeromekelleher** [comment]
LGTM; let's merge!

**Pull request merged**